backend/cmd/headlamp.go

```diff
@@ -829,6 +834,37 @@ func StartHeadlampServer(config *HeadlampConfig) {
 	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", config.port), handler)) //nolint:gosec
 }
 
+// CheckAndRemoveClusters checks for clusters marked as "dynamic_cluster"
+// and removes them if they are older than 10 minutes.
+func CheckAndRemoveClusters(kubeConfigStore kubeconfig.ContextStore) {
```

Comment:
Looks like the context is being removed irrespective of the last accessed time, if this is the required logic then it makes more sense to create a function like `AddContextWithLifespan` to make things a bit more organized and leverage the caches `SetWithTTL` functionality.

https://github.com/headlamp-k8s/headlamp/blob/41738dff59a9af0c3ad77529d5981f9b38169afb/backend/pkg/kubeconfig/contextStore.go#L10-L16

Reply:
Thanks for the idea. Made the necessary changes.